source/overview/architecture.rst

```diff
@@ -81,12 +81,60 @@ Moreover, search replicas are also supported to handle search queries.
 
 .. image:: ../images/architecture_high_availability.png
 
-Reference Architecture
+Reference Architectures
 ----------------------------------
 
+The following diagrams show the suggested architecture configurations for E20 enterprise deployments of Mattermost at different scales. These diagrams are meant as guidelines for typical Mattermost deployments. Hardware and infrastructure requirements can vary significantly based on usage and policies.
+
+There are two versions of each diagram. One is generalized while the other focuses on AWS.
+
+Each generalized diagram represents a full high availability deployment across all critical components. The proxy, database, file storage and Elasticsearch layers can be replaced by cloud services. Push proxy can be replaced by the Mattermost [HPNS](https://docs.mattermost.com/mobile/mobile-hpns.html).
```

Comment:
Please check earlier links in the doc on how this link should be formatted.

Reply:
Whoops, wrote it in Markdown first and missed this